### PR TITLE
Nuzhy/dataManager

### PR DESCRIPTION
**Pull request by Nuzhy-Deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/javascript/app/common/data_manager.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+class DataManager {
+    constructor() {
+        this.data = {
+            trade   : {},
+            purchase: {},
+            contract: {},
+        };
+    }
+
+    set(data, data_type, optional) {
```
Comment (prince-deriv) [inline src/javascript/app/common/data_manager.js]: it is not clear what the `optional` prop does

File: src/javascript/app/common/data_manager.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    clear(data_type) {
+        const oldValues = { ...this.data[data_type] };
+        this.data[data_type] = {};
+        const changeTypeMap = {
+            trade   : 'tradeChange',
+            purchase: 'purchaseChange',
+            contract: 'contractChange',
+        };
```
Comment (prince-deriv) [inline src/javascript/app/common/data_manager.js]: redeclared multiple times. Please move this above

File: src/javascript/app/pages/form/barrier-fields.jsx [hunk 3]
```diff
@@ -6,7 +6,8 @@ import {
 // eslint-disable-next-line import/no-unresolved
 import { StandaloneCircleInfoRegularIcon } from '@deriv/quill-icons/Standalone';
 import { useBarrierChange } from '../../hooks/events.js';
-import tradeManager from '../../common/trade_manager.js';
+// import tradeManager from '../../common/trade_manager.js';
```
Comment (prince-deriv) [inline src/javascript/app/pages/form/barrier-fields.jsx]: can remove this

File: src/javascript/app/common/data_manager.js (hunk 1 above)
Comment (Nuzhy-Deriv) [reply src/javascript/app/common/data_manager.js]: I changed the parameter name to optional_trigger. this is an optional param to set function to triggerBarrierChange,  triggerTimeChange only and to separate from tradeChange for not re-rendering everytime barrier, time change.

**Review by prince-deriv — APPROVED** [review]

**Pull request merged**